packages/theme/src/color-ramps/lib/find-color-with-constraints.ts

```diff
@@ -80,6 +81,13 @@ export function findColorMeetingRequirements(
 		} );
 	}
 
+	const contrastWithSeed = getContrast( reference, seed );
+
+	// Set the boundary based on the direction.
+	const mostContrastingL = direction === 'lighter' ? 1 : 0;
+	const mostContrastingColor = direction === 'lighter' ? WHITE : BLACK;
+	const highestContrast = getContrast( reference, mostContrastingColor );
```

Comment:
Moved this block few lines up because we use the `highestContrast` value to calculate the `deficit` return field.